Initial version of Google TTS bundle

dominicdesu opened the pull request:
This is the Google TTS bundle implementation as requested in #363.

It is using the JavaZoom JLayer library (http://www.javazoom.net/javalayer/javalayer.html) for mp3 output. The library is licenced under LGPL. I have just added the .jar so far. Do I have to add any references?

The bundle itself works like the other TTS bundles. Just use say() in a rule with activated bundle.
Language for text output can be defined in openhab.cfg (if not, English will be used).

Would be nice if the language could be passed in the say command (I didn't want to misuse voiceName or outputDevice for that purpose, even though they are not used by the GoogleTTS bundle). But that should probably handled in a seperate issue/pull request, if needed.

Closes #363

Signed-off-by: Dominic Lerbs <dominic.lerbs@gmail.com>

buildhive commented:
[openhab » openhab #2025](https://buildhive.cloudbees.com/job/openhab/job/openhab/2025/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[openhab » openhab #2027](https://buildhive.cloudbees.com/job/openhab/job/openhab/2027/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

dominicdesu commented:
Hi, this is the first time this has happened to me: Do I have to resolve the conflicts or is this done when the pull request is merged?

teichsta commented:
> Do I have to resolve the conflicts or is this done when the pull request is merged?

It would be great if you could resolve this conflict yourself.

Thanks, Thomas E.-E.

buildhive commented:
[openhab » openhab #2040](https://buildhive.cloudbees.com/job/openhab/job/openhab/2040/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[openhab » openhab #2041](https://buildhive.cloudbees.com/job/openhab/job/openhab/2041/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

teichsta commented on bundles/io/org.openhab.io.multimedia.tts.googletts/pom.xml:
```diff
@@ -0,0 +1,69 @@
+					<version>0.4</version>
+				</plugin>
+			</plugins>
+		</pluginManagement>
```
what is this maven external deps plugin is about?

dominicdesu replied:
Sorry, copied the pom from another bundle and didn't remove this unnecessary plugin.

teichsta commented on bundles/io/org.openhab.io.multimedia.tts.googletts/src/main/java/org/openhab/io/multimedia/internal/tts/TTSServiceGoogleTTS.java:
```diff
@@ -0,0 +1,153 @@
+	/** User agent for HTTP requests as requests without agent are ignored */
+	private static final String USER_AGENT = "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:34.0) "
+			+ "Gecko/20100101 Firefox/34.0";
+	private static final String GOOGLE_TRANSLATE_URL = "http://translate.google.com/translate_tts?tl=%s&q=%s";
```
it could be worth making this URL configurable

teichsta commented on bundles/io/org.openhab.io.multimedia.tts.googletts/src/main/java/org/openhab/io/multimedia/internal/tts/TTSServiceGoogleTTS.java:
```diff
@@ -0,0 +1,153 @@
+	private final GoogleTTSTextProcessor textProcessor = new GoogleTTSTextProcessor(MAX_SENTENCE_LENGTH);
+
+	public void activate() {
+		logger.debug("Activate");
```
please remove this statement or enhance it to something more context/meaningful

teichsta commented on bundles/io/org.openhab.io.multimedia.tts.googletts/src/main/java/org/openhab/io/multimedia/internal/tts/TTSServiceGoogleTTS.java:
```diff
@@ -0,0 +1,153 @@
+	}
+
+	public void deactivate() {
+		logger.debug("Deactivate");
```
please remove this statement or enhance it to something more context/meaningful

teichsta commented on bundles/io/org.openhab.io.multimedia.tts.googletts/src/main/java/org/openhab/io/multimedia/internal/tts/TTSServiceGoogleTTS.java:
```diff
@@ -0,0 +1,153 @@
+				} catch (IOException e) {
+					logger.debug("Error while closing InputStream", e);
+				}
+			}
```
you could use IOUtils.closeQuitely() instead

teichsta commented:
Hi @dominicdesu, thanks a lot for this contribution. Please find some review comments inline. Best, Thomas E.-E.

dominicdesu commented:
Hi @teichsta, thank you for the thorough code review. It's a great way to learn!
I have addressed your comments and commited the changes. Please let me now if there are still open issues/questions!

buildhive commented:
[openhab » openhab #2043](https://buildhive.cloudbees.com/job/openhab/job/openhab/2043/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)
